src/Cmfd.cpp

```diff
@@ -418,26 +395,26 @@ void Cmfd::computeDs(int moc_iteration) {
           else{
 
             /* Set properties for cell next to Surface */
-            if (surface == 0) {
+            if (surface == SURFACE_X_MIN) {
```

Comment:
Again, it appears that irregardless of the value of ``surface``, the ``next_length_perpen`` variable is set to ``_cell_height``.